fluentd/aws-ecs/image/fargate.rb

```diff
@@ -0,0 +1,48 @@
+def start
+  super
+  # This is the first method to be called when it starts running
+  # Use it to allocate resources, etc.
+  require 'json'
+  # get docker metadata file
+  `curl -s ${ECS_CONTAINER_METADATA_URI_V4}/task > /var/log/docker-metadata.json`
+  # create cache
+  @metadata_cache = {}
+  # populate cache
+  json = JSON.parse(File.read('/var/log/docker-metadata.json'))
+  json['Containers'].each do |container|
+    docker_id = container['DockerId']
+    @metadata_cache[docker_id] = container
+  end
+end
+
+def shutdown
+  super
+  # This method is called when Fluentd is shutting down.
+  # Use it to free up resources, etc.
+  @metadata_cache = nil
+end
+
+def filter(tag, time, record)
+  # This method implements the filtering logic for individual filters
+  # Get docker id
+  if record.key?('container_id') then
+    docker_id = record['container_id']
+    metadata = @metadata_cache[docker_id]
+    if metadata != nil then
+      # Create the docker field
+      record['docker'] = {
+        'id' => metadata['DockerId'],
+        'name' => metadata['Name'],
+        'container_hostname' => metadata['DockerName'],
+        'image' => metadata['Image'],
+        'image_id' => metadata['ImageID'],
+        'labels' => metadata['Labels']
+      }
+      record.delete('container_name')
+      record.delete('container_id')
```

Comment:
I think we should remove the two other fields that gets generated in the docker metadata:
- ecs_task_arn
- ecs_task_definition

Reply:
also i think docker.name and docker.container_hostname in fargate will always be the same name so we don't need both.

Reply:
task_definition I think is helpful due to the version. But what ever you think.

Reply:
we also have the version in another docker label so no need for the ecs metadata one.

Reply:
Removed the duplications.